### PR TITLE
CLN: BlockManager.apply

### PR DESCRIPTION
**Pull request by jbrockmendel**
Working on arithmetic, we're going to be sending some functions through BlockManager.apply.  This is just cleaning that up a bit as a preliminary.

### CODE REVIEW

File: pandas/core/internals/managers.py [hunk 1]
```diff
@@ -432,10 +418,8 @@ def apply(
             result_blocks = _extend_blocks(applied, result_blocks)
 
         if len(result_blocks) == 0:
-            return self.make_empty(axes or self.axes)
-        bm = self.__class__(
-            result_blocks, axes or self.axes, do_integrity_check=do_integrity_check
-        )
+            return self.make_empty(self.axes)
+        bm = self.__class__(result_blocks, self.axes, do_integrity_check=False)
```
Comment (topper-123) [inline pandas/core/internals/managers.py]: The original had ``axes or self.axes``. could we not risk that an ``axes``parameter is in kwargs?

Would it be possible ti be explicit about what parameter thid accepts, i.e. not use kwargs?

File: pandas/core/internals/managers.py [hunk 2]
```diff
@@ -772,13 +756,15 @@ def copy(self, deep=True):
         # this preserves the notion of view copying of axes
         if deep:
             if deep == "all":
-                copy = lambda ax: ax.copy(deep=True)
+                copy_func = lambda ax: ax.copy(deep=True)
```
Comment (topper-123) [inline pandas/core/internals/managers.py]: Could you make this a named function?  Same below.

**Review by topper-123 — CHANGES_REQUESTED** [review]

File: pandas/core/internals/managers.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/internals/managers.py]: > The original had axes or self.axes. could we not risk that an axesparameter is in kwargs?

The only case that passed axes was copy, which this changes to not do.

> Would it be possible ti be explicit about what parameter this method accepts, i.e. not use kwargs?

I'm hopeful we'll get there before long

**Review by jreback — APPROVED** [review]
lgtm, comment by @topper-123

**Comment by jbrockmendel** [comment]
rebased+green

File: pandas/core/generic.py [hunk 3]
```diff
@@ -5664,7 +5664,7 @@ def _to_dict_of_blocks(self, copy=True):
             for k, v, in self._data.to_dict(copy=copy).items()
         }
 
-    def astype(self, dtype, copy=True, errors="raise"):
+    def astype(self, dtype, copy: bool_t = True, errors: str = "raise"):
```
Comment (jreback) [inline pandas/core/generic.py]: did you mean to use bool_t here?

Comment (jbrockmendel) [reply pandas/core/generic.py]: Yes.  We can't use `bool` here because NDFrame defines bool, see note on L142

Comment (jreback) [reply pandas/core/generic.py]: ok, interesting

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @jbrockmendel

**Pull request merged**